include/zephyr/sys/sys_io.h

```diff
@@ -265,6 +265,18 @@ typedef uintptr_t mem_addr_t;
  * @param mask the bit mask of a 32 bits data to set
  */
 
+/**
+ * @fn static inline void sys_modify_bits(mem_addr_t addr, unsigned int mask, unsigned int value)
+ * @brief Modify the bits from addr to the values defined by @p mask and @p value
+ *
+ * This functions modifies the designated bits from addr by clearing mask before setting value.
+ * It is equivalent to calling sys_clear_bits then sys_set_bits, but addr is modified only once.
```

Comment:
I would suggest rewording this to something like the following to better describe what it does, not how it does it ...

This function modifies the subset of bits defined by @p mask at address @p addr
to match the corresponding subset of bits in @p value. It is equivalent to calling
sys_clear_bits() and then sys_set_bits(), except that the data at the specified
address is only modified once.